File: Python/Split_a_String_in_Balanced_Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        output = counter = 0
        
        for char in s:
            
            if char == 'L':
                counter -= 1
            else:
                counter += 1
            
            if counter == 0:
                output +=1
                
        return output

File: C++/Split_a_String_in_Balanced_Strings.cpp
using namespace std;
class Solution {
public:
    int balancedStringSplit(string s) {
        int balanced = 0;
        int numBalanced = 0;
        for(int i = 0; i < s.size(); i++){
            if (s[i] == 'R')
                balanced +=1;
            else
                balanced -=1;
            if (balanced == 0)
                numBalanced += 1;
        }
        return numBalanced;
    }
};